driver/mseloss_driver.hpp

```diff
@@ -0,0 +1,590 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef GUARD_MIOPEN_MSELOSS_DRIVER_HPP
+#define GUARD_MIOPEN_MSELOSS_DRIVER_HPP
+
+#include "driver.hpp"
+#include "miopen/errors.hpp"
+#include "miopen/miopen.h"
+#include "miopen/tensor.hpp"
+#include "miopen/tensor_view_utils.hpp"
+#include "random.hpp"
+#include "../test/verify.hpp"
+#include "tensor_driver.hpp"
+#include <cstddef>
+#include <cstdlib>
+#include <iostream>
+#include <memory>
+#include <vector>
+#include "../test/tensor_holder.hpp"
+#include "tensor_view.hpp"
+#include "timer.hpp"
+
+inline std::vector<std::string> split(const std::string& s, char delim)
+{
+    std::vector<std::string> elems;
+    std::stringstream ss(s);
+    std::string item;
+    while(std::getline(ss, item, delim))
+    {
+        elems.push_back(item);
+    }
+    return elems;
+}
+
+template <typename T>
+inline std::vector<T> ComputeStrides(std::vector<T> input, bool contiguous)
+{
+    if(!contiguous)
+        std::swap(input.front(), input.back());
+    std::vector<T> strides(input.size());
+    strides.back() = 1;
+    for(int i = input.size() - 2; i >= 0; --i)
+        strides[i] = strides[i + 1] * input[i + 1];
+    if(!contiguous)
+        std::swap(strides.front(), strides.back());
+    return strides;
+}
+
+template <typename Tgpu, typename Tref>
+void mloMSELossForwardRunHost(miopenTensorDescriptor_t inputDesc,
+                              miopenTensorDescriptor_t targetDesc,
+                              miopenTensorDescriptor_t outputDesc,
+                              const Tgpu* input,
+                              const Tgpu* target,
+                              Tref* output,
+                              float divisor)
+{
+    const int local_size = 256;
+
+    tensor_view_t<5> I_tv = miopen::get_inner_expanded_tv<5>(miopen::deref(inputDesc));
+    tensor_view_t<5> T_tv = miopen::get_inner_expanded_tv<5>(miopen::deref(targetDesc));
+
+    int64_t gid = 0;
+    auto size   = miopen::deref(inputDesc).GetElementSize();
+    auto ref_workspace =
+        std::vector<Tref>(size + ((size / local_size) + 1) * local_size, static_cast<Tref>(0.0f));
+
+    while(true)
+    {
+        size_t n0123 = gid / I_tv.size[4], n4 = gid % I_tv.size[4];
+        size_t n012 = n0123 / I_tv.size[3], n3 = n0123 % I_tv.size[3];
+        size_t n01 = n012 / I_tv.size[2], n2 = n012 % I_tv.size[2];
+        size_t n0 = n01 / I_tv.size[1], n1 = n01 % I_tv.size[1];
+
+        if(!(n0 < I_tv.size[0]))
+            break;
+
+        size_t Iidx = I_tv.get_tensor_view_idx({n0, n1, n2, n3, n4});
+        size_t Tidx = T_tv.get_tensor_view_idx({n0, n1, n2, n3, n4});
+
+        ref_workspace[gid] = static_cast<Tref>((input[Iidx] - target[Tidx]) *
```

Comment:
```suggestion
        *output += static_cast<Tref>((input[Iidx] - target[Tidx]) *
```

Reply:
Should be fixed in 1a41c9dfa729be6d608e3a639595e95902abd880

Reply:
May I ask you to do the same for the same code in the tests directory?

Reply:
Done in 09b16d2